File: src/bfort/testing2/testref-old/thandle.f90
      module f90header
      interface
      function KSPSetNormType(ksp, normtype )
       KSP ksp ! KSP
       KSPNormType normtype ! KSPNormType
       PetscErrorCode  KSPSetNormType ! PetscErrorCode 
       end function KSPSetNormType
      function KSPSetSupportedNorm(ksp, normtype, pcside, priority )
       KSP ksp ! KSP
       KSPNormType normtype ! KSPNormType
       PCSide pcside ! PCSide
       PetscInt priority ! PetscInt
       PetscErrorCode  KSPSetSupportedNorm ! PetscErrorCode 
       end function KSPSetSupportedNorm
      function KSPGetNormType(ksp, normtype )
       KSP ksp ! KSP
       KSPNormType normtype ! KSPNormType
       PetscErrorCode  KSPGetNormType ! PetscErrorCode 
       end function KSPGetNormType
      function KSPSetOperators(ksp, Amat, Pmat )
       KSP ksp ! KSP
       Mat Amat ! Mat
       Mat Pmat ! Mat
       PetscErrorCode  KSPSetOperators ! PetscErrorCode 
       end function KSPSetOperators
      function KSPGetOperators(ksp, Amat, Pmat )
       KSP ksp ! KSP
       Mat Amat ! Mat
       Mat Pmat ! Mat
       PetscErrorCode  KSPGetOperators ! PetscErrorCode 
       end function KSPGetOperators
      function KSPCreate(comm, inksp )
       MPI_Comm comm ! MPI_Comm
       KSP inksp ! KSP
       PetscErrorCode  KSPCreate ! PetscErrorCode 
       end function KSPCreate
      subroutine LongArgList(inksp, thisisthefirstlongargument,         &
     &thisisthesecondlongargument, thisisusedforanoutput,               &
     &alongnamedscalar, alongnamedvector, thefinallongargumentname )
       KSP inksp ! KSP
       Mat thisisthefirstlongargument ! Mat
       Mat thisisthesecondlongargument ! Mat
       integer thisisusedforanoutput ! int
       double precision alongnamedscalar ! double
       double precision alongnamedvector ! double
       integer thefinallongargumentname ! int
      
       end subroutine LongArgList
      end interface
      end module f90header

File: src/bfort/testing2/thandle.c
/* Test of support for other handle options, using the PETSc examples.
 */

#include <petsc/private/kspimpl.h>      /*I "petscksp.h" I*/
#include <stdlib.h>                     /*I <stdlib.h> I*/

/*@
   KSPSetNormType - Sets the norm that is used for convergence testing.

$ Here are some verbatim comments to skip
Args:
+ foo
. bar
- foobar

@*/
PetscErrorCode  KSPSetNormType(KSP ksp, KSPNormType normtype)
{}

/*@
   KSPSetSupportedNorm - Sets a norm and preconditioner side supported by a KSP

@*/
PetscErrorCode KSPSetSupportedNorm(KSP ksp,KSPNormType normtype,PCSide pcside,PetscInt priority)
{}

/*@
   KSPGetNormType - Gets the norm that is used for convergence testing.
@*/
PetscErrorCode  KSPGetNormType(KSP ksp, KSPNormType *normtype)
{}

/*@
   KSPSetOperators - Sets the matrix associated with the linear system
   and a (possibly) different one associated with the preconditioner.
@*/
PetscErrorCode  KSPSetOperators(KSP ksp,Mat Amat,Mat Pmat)
{}

/*@
   KSPGetOperators - Gets the matrix associated with the linear system
   and a (possibly) different one associated with the preconditioner.
@*/
PetscErrorCode  KSPGetOperators(KSP ksp,Mat *Amat,Mat *Pmat)
{}

/*@
   KSPCreate - Creates the default KSP context.
@*/
PetscErrorCode  KSPCreate(MPI_Comm comm,KSP *inksp)
{}

/*@
   LongArgList - Routine with a long argument list to test Fortran 90
   output routines

  @*/
void LongArgList(KSP inksp, Mat thisisthefirstlongargument,
		 Mat thisisthesecondlongargument, int *thisisusedforanoutput,
		 double alongnamedscalar, double *alongnamedvector,
		 int thefinallongargumentname)
{
}
